WooCommerce/Classes/ViewRelated/Orders/Order Creation/Synchronizer/LocalOrderSynchronizer.swift

```diff
@@ -93,4 +95,13 @@ private extension LocalOrderSynchronizer {
             }
             .assign(to: &$order)
     }
+
+    /// Creates a new input with a random ID when the given ID is `.zero`.
+    ///
+    func replaceInputWithLocalIDIfNeeded(_ input: OrderSyncProductInput) -> OrderSyncProductInput {
+        guard input.id == .zero else {
+            return input
+        }
+        return input.updating(id: Int64(UUID().uuidString.hashValue))
+    }
```

Comment:
Why do we need non-zero id in `LocalOrderSynchronizer`?

Reply:
`LocalOrderSynchronizer ` never used `.zero` ids, They were autogenerated in the at the init.

https://github.com/woocommerce/woocommerce-ios/blob/e96b706fafd25c55bc958de6dfd4692c914888b7/WooCommerce/Classes/ViewRelated/Orders/Order%20Creation/Synchronizer/OrderSynchronizer.swift#L22

In this PR we move the autogeneration to be a bit more explicit.